### PR TITLE
Refactor FXIOS-6021 [v120] Swiftlint: remove force try from disabled rules

### PR DESCRIPTION
**Pull request by atfelix**
## :scroll: Tickets
[Jira ticket](https://mozilla-hub.atlassian.net/browse/FXIOS-6021)
[Github issue](https://github.com/mozilla-mobile/firefox-ios/issues/13666)

## :bulb: Description
I remove `force_try` from the `.swiftlint.yml`'s disabled rules and fixed any related errors.

## :pencil: Checklist
You have to check all boxes before merging
- [x] Filled in the above information (tickets numbers and description of your work)
- [x] Updated the PR name to follow our [PR naming guidelines](https://github.com/mozilla-mobile/firefox-ios/wiki/Pull-Request-Naming-Guide)
- [x] Wrote unit tests and/or ensured the tests suite is passing
- [x] When working on UI, I checked and implemented accessibility (minimum Dynamic Text and VoiceOver)
- [x] If needed I updated documentation / comments for complex code and public methods

### CODE REVIEW

File: Tests/ClientTests/Mocks/MockProfile.swift
Comment (atfelix) [inline Tests/ClientTests/Mocks/MockProfile.swift:1]: `MockProfile` is used in numerous files / tests.  I considered altering `MockProfile.init` to be a throwing initializer, which would require changing all calling tests (they would need to be a `throws`test).  I prefer throwing tests with throwing initializers.  But before changing `MockProfile.init` as such, I wanted the opinion of the Mozilla maintainers.

**Comment by adudenamedruby** [comment]
Sorry. WIll get to review this today or tomorrow... been tied up with a few things.

**Comment by adudenamedruby** [comment]
Hey, BitRise is failing on configuring the build because of threshold warnings. There was a weird issue on `main` that's now been fixed. Would you mind rebasing against `main` so we get those fixes in and then I can try to rerun BitRise.

**Comment by adudenamedruby** [comment]
Sorry about this, but you'll need to rebase once more becasue we've switch over to arm64 builds and that change is in the latest `bitrise.yaml` so BR will fail without it. Hopefully this is the last time. lol

**Comment by adudenamedruby** [comment]
@yoanarios - this just needs help getting over the line, but I'm heading out for vacation. If you could take over. Things look fine to me, but BR is failing for some reason.

File: Client/Application/UITestAppDelegate.swift
Comment (isabelrios) [inline Client/Application/UITestAppDelegate.swift:63]: Not sure about these changes here, some tests that have been really stable are failing in this PR. Please let me take a look at this PR before landing

Comment (isabelrios) [reply Client/Application/UITestAppDelegate.swift:63]: Looks like the launch arguments are not working as expected in some scenarios with these changes and so the tests fail... 
For example, for this test: testFirstRunTour the Onboarding is not shown but it should be there since we have a setUp method with specific arguments.

Comment (yoanarios) [reply Client/Application/UITestAppDelegate.swift:63]: thank you @isabelrios I will take a look into the failure I feel it can be related to one of the unwrapping changes and it could be failing to load the onboarding flow

Comment (isabelrios) [reply Client/Application/UITestAppDelegate.swift:63]: thanks @yoanarios let me know if you need more details about the failures

Comment (yoanarios) [reply Client/Application/UITestAppDelegate.swift:63]: I checked the onboarding test and I can reproduce the failure locally.
Note: The test are passing on main and when running the app the onboarding is shown properly so the failure seems related to the test scheme only. I couldn't fin any evident cause at first glance

Comment (isabelrios) [reply Client/Application/UITestAppDelegate.swift:63]: The issue is that the onboarding should be shown in the test but it is not. We skip the onboarding in all tests with a specific launch argument. In this case we are overwriting the set up so that the onboarding is shown in all the tests in this tes suite. The launch arguments are managed in this file, that’s why my feeling is that somehow changing this force try are breaking something with the way the app is launched for testing

Comment (atfelix) [reply Client/Application/UITestAppDelegate.swift:63]: One way we could test is by disabling the `force_try` rule for the lines in the file and revert the rest of these changes.  I'm not sure how force try would change the launch arguments unless there is some recovery mechanism in the test plans.  But we can see if you're right with such a change

Comment (isabelrios) [reply Client/Application/UITestAppDelegate.swift:63]: I will be doing some changes to try to isolate the issue. Thanks for your help

File: Providers/Profile.swift
Comment (isabelrios) [inline Providers/Profile.swift:465]: @yoanarios @atfelix after some investigation, the issue with the tests comes from the changes in this file. For some reason this is breaking the test in the Onboarding test suite and the Search tests that you see failing in this PR.

I undid these changes and all the tests pass.

Comment (atfelix) [reply Providers/Profile.swift:465]: Thank you for finding the issue.

Are the test failures occurring locally or only on CI?  If they're failing locally, I can investigate after work (5 PM Toronto time).

Comment (isabelrios) [reply Providers/Profile.swift:465]: They happen locally too.
If you select SmokeXCUITests Test Plan, the OnboardingTests suite will be enable, run it and you will see the failures.
Same for the SearchTests suite, but using the Smoketest4 as the TestPlan

Hope that helps. Thanks for taking the time to look at this.

Comment (atfelix) [reply Providers/Profile.swift:465]: I think I have a fix for this issue.  I believe (but it's hard to know for sure) that the `clear` in `BrowserProfile.init` and where I put the `directory` do block is causing the issue.  I have moved the `directory` to a lazy variable that crashes via a `fatalError` in the `catch` block (same behaviour as current production).  This move seems to fix the issue but it's difficult to say as running the tests and getting consistent results seems difficult (at least locally).

Please let me know if there are any other CI issues.

**Comment by isabelrios** [comment]
Looks like Bitrise is ✅ with latest changes 🎉

**Review by yoanarios — APPROVED** [review]
Great work @atfelix and thank you for your contribution

**Comment by yoanarios** [comment]
> Looks like Bitrise is ✅ with latest changes 🎉

Thank you @isabelrios for your support on this issue

**Pull request merged**
